src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,27 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationValidationException("User cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationValidationException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new
+                    RegistrationValidationException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationValidationException("Age does not meet the condition");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
It's better to move the magic number '18' to a constant field with an informative name to avoid hardcoding values directly in the code.